ctaplot/gammaboard/gammaboard.py

```diff
@@ -80,19 +80,22 @@ def load_trig_events(experiment, experiments_directory):
     result_files = find_data_files(experiment, experiments_directory)
     trig_energies = []
     for file in result_files:
-        try:
-            result_file = tables.open_file(file)
-        except Exception as e:
-            print('Could not open data file {}'.format(file))
-            print(e)
-        else:
+        if guess_particle_type_from_file(file) == GAMMA_ID:
             try:
-                if guess_particle_type_from_file(file) == GAMMA_ID:
-                    result_file.close()
+                with tables.open_file(file) as f:
```

Comment:
I think the code could be simplified by using tables earlier:

```
from lstchain.io.io import dl1_params_lstcam_key
from astropy.table import Table, vstack

for file in result_files:
    if guess_particle_type_from_file(file) == GAMMA_ID:
        try:
            dl1_params = Table.read(file, path= dl1_params_lstcam_key)
            trig_energies.append([dl1_params['mc_energy'][dl1_params['obs_id'] == obs_id] for obs_id in set(dl1_params['obs_id'])])
            
        except OSError:
            try:
                 trig_energies.append(Table.from_pandas(pd.read_hdf(file, key='triggered_events')))
                
             except KeyError as e:
                print("Cannot load the number of triggered events for experiment {} file".format(experiment))
                return None
energies = vstack(trig_energies)
energies.rename_column('mc_energy', 'mc_trig_energies')
return energies['mc_trig_energies'] * u.TeV
```

Reply:
of course I have not tested this :D

Reply:
done as discussed